### PR TITLE
Add git ignore file for IDEA project files

### PR DESCRIPTION
**Pull request by semanticoverload**
adding a ```.gitignore``` file so that inteliJ project files don't get added to the repo

### CODE REVIEW

**Comment by semanticoverload** [comment]
@aunueget if you want to use inteliJ for this I can update so that the project related files are included in the repo and user specific files are ignored similar to [this gitignore](https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore)

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#inteliJ file type
+*.iml
+#inteliJ directory
+.idea/
```
Comment (aunueget) [inline .gitignore:4]: We may want to add some of these for java as well:
Reference:https://github.com/github/gitignore/blob/master/Java.gitignore
```
# Compiled class file
*.class

# Log file
*.log

# BlueJ files
*.ctxt

# Mobile Tools for Java (J2ME)
.mtj.tmp/

# Package Files #
*.jar
*.war
*.ear
*.zip
*.tar.gz
*.rar

# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
hs_err_pid*
```

**Review by aunueget — CHANGES_REQUESTED** [review]
Do you think we should add some of these to the ignore?

**Comment by semanticoverload** [comment]
Looks like the newer repo has a few more items. I prefer making changes as needed. I'm sure we will have other exceptions if we use it or edit it a little bit or if we add a build tool.

**Pull request closed without merging**